### PR TITLE
[Merged by Bors] - chore: add docstrings to some tactic items

### PR DESCRIPTION
**Pull request by grunweg**
Remove a couple of nolint entries.

The `cases'` docstring has essentially been copied from the module docstring; the other entries are new.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/Cases.lean [hunk 1]
```diff
@@ -101,6 +123,28 @@ elab (name := induction') "induction' " tgts:(Parser.Tactic.casesTarget,+)
...
+  · exact Or.inr hp
+  · exact Or.inl hq
+
+example (h : p ∨ q) : q ∨ p := by
+  rcases h with hp | hq
+  · exact Or.inr hp
+  · exact Or.inl hq
+```
+
+Prefer `cases` or `rcases` when possible, because these tactics promote structured proofs.
```
Comment (joneugster) [inline Mathlib/Tactic/Cases.lean:146]: There is [rcases vs obtain](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Is.20cases'.20deprecated.3F/near/433757205), so I wonder if `rcases` shouldn't be mentioned (in favour of `obtain`). But there seems to be consent there to leave that for a future refactor tool...

File: Mathlib/Tactic/Rename.lean [hunk 2]
```diff
@@ -14,6 +14,8 @@ namespace Mathlib.Tactic
 
 open Lean Elab.Tactic Meta
 
+/-- A parser for a single rename to perform in the `rename` tactic:
+these should have the form `h => hnew` (describing a rename of a hypothesis `h` to `hnew`) -/
```
Comment (joneugster) [inline Mathlib/Tactic/Rename.lean:18]:
```suggestion
these should have the form `h => hnew` (describing a rename of a hypothesis `h` to `hnew`). -/
```

**Review by joneugster — APPROVED** [review]
Thanks for doing this!

maintainer merge

File: Mathlib/Tactic/Cases.lean [hunk 3]
```diff
@@ -68,6 +67,29 @@ def ElimApp.evalNames (elimInfo : ElimInfo) (alts : Array ElimApp.Alt) (withArg
   pure subgoals
 
 open private getElimNameInfo generalizeTargets generalizeVars from Lean.Elab.Tactic.Induction
+/-- The `induction'` tactic is similar to the `induction` tactic in Lean 4 core,
+but with slightly different (such as, no requirement to name the constructors).
```
Comment (semorrison) [inline Mathlib/Tactic/Cases.lean:71]:
```suggestion
but with slightly different syntax (such as, no requirement to name the constructors).
```

**Comment by semorrison** [comment]
bors d+

**Comment by grunweg** [comment]
bors merge

**Pull request closed without merging**